lib/GLPI/Agent/SNMP/MibSupport/Ubnt.pm

```diff
@@ -32,7 +58,56 @@ sub getSerial {
 sub getMacAddress {
     my ($self) = @_;
 
-    return $self->get(ubntWlStatApMac);
+    return getCanonicalMacAddress($self->get(ubntWlStatApMac));
+}
+
+sub run {
+    my ($self) = @_;
+
+    my $device = $self->device
+        or return;
+
+    # Get list of device ports (e.g. raX, raiX etc.)
+    my $ports = $device->{PORTS}->{PORT};
+
+    # Get list of SSID
+    my $unifiVapEssidValues = $self->walk(unifiVapEssid) || {};
+    # Get list of Radios (e.g. ra0, rai0 etc.)
+    my $unifiVapNameValues = $self->walk(unifiVapName) || {};
+    # The list of Radios is co-related to the list of SSIDs
+    # $unifiVapNameValues->{0} = ra0
+    # $unifiVapEssidValues->{0} = <SSID>
+
+    foreach my $port (keys(%$ports)) {
+        # For each device Radio port (raX, raiX etc.)
+        # If you have more than one SSID there will also be more raX, raiX for each SSID.
+        my $ifdescr = $device->{PORTS}->{PORT}->{$port}->{IFDESCR};
+        next unless defined($ifdescr) && $ifdescr =~ /^ra/;
+
+        foreach my $index (keys(%$unifiVapNameValues)) {
+            # Compares the device's current radio port name to the AP's radio list (e.g. raX eq raX)
+            if ($ifdescr eq $unifiVapNameValues->{$index}) {
+                # Defines the port alias with the name of the radio (e.g. raX)
+                $device->{PORTS}->{PORT}->{$port}->{IFALIAS} = $ifdescr;
+                # Replaces the radio port name with its respective <SSID>
+                $device->{PORTS}->{PORT}->{$port}->{IFNAME} = getCanonicalString($unifiVapEssidValues->{$index});
+
+                # raX and raiX are the network interfaces for the 2.4GHz and 5GHz radios respectively
+                if ($ifdescr =~ m/^ra\d+$/) {
+                    $device->{PORTS}->{PORT}->{$port}->{IFNAME} .= " (2.4GHz)";
+                } elsif ($ifdescr =~ m/^rai\d+$/) {
+                    $device->{PORTS}->{PORT}->{$port}->{IFNAME} .= " (5GHz)";
+                }
+
+                # Replaces the port iftype from "Ethernet" (6) to "WiFi" (71)
+                if($device->{PORTS}->{PORT}->{$port}->{IFTYPE} == 6) {
+                    $device->{PORTS}->{PORT}->{$port}->{IFTYPE} = 71
+                }
```

Comment:
You again forgot a space after the `if` statement, not a big deal, but this is just to be coherent ;-)

Anyway why putting this here and not just after the test on line 85 ? Is description starting with "ra" not a sufficient condition to change the default type ?